BtatesCraft/src/scenes/GameScene.h

```diff
@@ -34,10 +34,10 @@ namespace Scene
 		glm::mat4 ProjectionMatrix;
 		glm::mat4 MVP;
 		unsigned int Seed;
-		std::array<std::unique_ptr<Chunck>, WORLD_AREA > chuncks;
 		std::unique_ptr<Camera> camera;
 		std::unique_ptr<BlockPicker> Picker;
 		std::unique_ptr<BlockRenderManger> BlockRenderer;
+		std::array<std::unique_ptr<Chunck>, WORLD_AREA > chuncks;
```

Comment:
The array of chunks must live longer than the instance of BlockRenderManager.
See https://github.com/MohabMohamed/BtatesCraft/blob/master/BtatesCraft/src/Chunck.cpp#L43 : the chunk will access the block renderer when it is destroyed. The block renderer could instead also be a shared_ptr for that case